File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int ec) {
        int mx=0;
        int n=candies.size();
        for(auto v: candies){
            if(mx<v) mx=v;
        }
        vector<bool>ans(n,false);
        for(int i=0;i<n;++i){
            if(candies[i]+ec>=mx) ans[i]=true;
        }
        return ans;
    }
};

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.py
class Solution:
    def kidsWithCandies(self, c: List[int], ec: int) -> List[bool]:
        ans=[False]*(len(c))
        h=max(c)
        for i in range(len(c)):
            if c[i]+ec>=h:
                ans[i]=True
        return ans